airflow-core/src/airflow/models/deadline.py

```diff
@@ -366,6 +367,95 @@ def _evaluate_with(self, *, session: Session, **kwargs: Any) -> datetime:
 
             return _fetch_from_db(DagRun.queued_at, session=session, **kwargs)
 
+    @dataclass
+    class AverageRuntimeDeadline(BaseDeadlineReference):
+        """A deadline that calculates the average runtime from past DAG runs."""
+
+        DEFAULT_LIMIT = 10
+        limit: int
+        min_runs: int | None = None
```

Comment:
If we verify the user input and apply defaults in the Task SDK definition, then I think we should just be able to assume that min_runs is provided and is valid
```suggestion
        min_runs: int
```
(and also remove `__post_init__` if we go ahead with this)